server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -319,30 +324,29 @@ ClusterState addComponentTemplate(
     }
 
     @Nullable
-    private static String wrapMappingsIfNecessary(@Nullable String mappings, NamedXContentRegistry xContentRegistry) throws Exception {
+    private static CompressedXContent wrapMappingsIfNecessary(@Nullable CompressedXContent mappings, NamedXContentRegistry xContentRegistry)
+        throws Exception {
```

Comment:
I think this can't even throw to begin with. We seem to only be parsing mappings from templates in this method that we would have parsed elsewhere before (and if we didn't we should). I think rather than messing with this method's parsing validation any more, I'd prefer to make it go away in a follow-up that we have lined up anyway around deduplicating mappings. I'll see if I can add the mapper exception back in an earlier spot though :)